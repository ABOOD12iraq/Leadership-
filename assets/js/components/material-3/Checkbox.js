/**
 * Checkbox component.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@material/web/checkbox/checkbox';
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { useEffect, useRef, useState } from '@wordpress/element';

export default function Checkbox( {
	checked,
	indeterminate,
	disabled,
	name,
	value,
	onChange,
	onKeyDown,
} ) {
	console.log( 'Checkbox render. checked:', checked );
	const ref = useRef( null );

	// Use this state in a key prop on the web component to force a rerender.
	// This is a workaround for the fact the web component doesn't update visually when the checked property is changed.
	const [ index, setIndex ] = useState( 0 );

	useEffect( () => {
		const { current } = ref;

		const click = ( event ) => {
			console.log( 'click', event );

			// Prevent default and manually dispatch a change event, in order to retain checkbox state and use as a controlled input.
			event.preventDefault();

			if ( disabled ) {
				return;
			}

			current.checked = current.checked ? undefined : true;

			current.dispatchEvent(
				new Event( 'change', {
					bubbles: true,
					composed: true,
				} )
			);
		};

		const change = ( event ) => {
			console.log( 'change', event );

			// Preventing default behaviour has no effect here.
			// event.preventDefault();

			onChange?.( event );

			setIndex( index + 1 );
		};

		// The click event is triggered by both mouse and keyboard entry (space key) on Chrome, need to confirm other browsers.
		current?.addEventListener( 'click', click );
		current?.addEventListener( 'change', change );

		// The 'action' event is dispatched by the ActionElement base class of the Material checkbox. See ActionElement source for details.
		// current?.addEventListener( 'action', cancelEvent );

		return () => {
			current?.removeEventListener( 'click', click );
			current?.removeEventListener( 'change', change );
		};
	}, [ checked, disabled, index, onChange ] );

	// TODO: Change theme colour to match expected ?

	return (
		<md-checkbox
			key={ `checkbox-${ name }-${ index }` }
			ref={ ref }
			checked={ checked }
			indeterminate={ indeterminate }
			disabled={ disabled }
			name={ name }
			value={ value }
		></md-checkbox>
	);
}

/**
 * Props determined by examining the Material 3 web component's source code.
 *
 * See @property definitions and event handlers in the source to identify which props are needed.
 */

Checkbox.propTypes = {
	// Fundamental checkbox attributes.
	checked: PropTypes.bool,
	indeterminate: PropTypes.bool,
	disabled: PropTypes.bool,
	name: PropTypes.string,
	value: PropTypes.string,

	// Accessibility attributes.
	ariaLabel: PropTypes.string,
	ariaLabelledBy: PropTypes.string,
	ariaDescribedBy: PropTypes.string,
	reducedTouchTarget: PropTypes.bool,

	// Event handlers.
	onChange: PropTypes.func,
	onKeyDown: PropTypes.func,
};
