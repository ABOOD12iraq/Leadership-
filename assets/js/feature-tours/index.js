/**
 * Feature Tours.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';

const allTrafficWidget = {
	slug: 'allTrafficWidget',
	contexts: [ VIEW_CONTEXT_DASHBOARD ],
	version: '1.25.0',
	checkRequirements: async ( registry ) => {
		// Here we need to wait for the underlying selector to be resolved before selecting `isModuleConnected`.
		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
		if ( ! registry.select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
			return false;
		}
		return true;
	},
	steps: [
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__user-count-chart',
			title: __( "It's now easier to see your site's traffic at a glance", 'google-site-kit' ),
			content: __( 'Check the trend graph to see how your traffic changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
			title: __( 'See where your visitors come from', 'google-site-kit' ),
			content: __( 'Click on the chart slices to see how each segment has changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-header__date-range-selector-menu',
			title: __( 'Check how your traffic changed since you last looked', 'google-site-kit' ),
			content: __( 'Select a time frame to see the comparison with the previous time period', 'google-site-kit' ),
		},
	],
};

// Ordered tours.
export default [
	allTrafficWidget,
];
