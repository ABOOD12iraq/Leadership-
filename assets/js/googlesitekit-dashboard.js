/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { clearWebStorage } from './util';
import Root from './components/Root';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		// Make a separate API request since we cannot use selectors
		// outside the Root component.
		const { authenticated: isAuthenticated } = await API.get(
			'core',
			'user',
			'authentication',
			{},
			{ useCache: false }
		);

		const { setupModuleSlug } = renderTarget.dataset;
		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = isAuthenticated
				? VIEW_CONTEXT_DASHBOARD
				: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY;
		}
		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
