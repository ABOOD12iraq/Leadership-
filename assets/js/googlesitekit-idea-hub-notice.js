/**
 * Idea Hub Notice (for Block Editor) entrypoint.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { getItem, setItem } from './googlesitekit/api/cache';

const shownNotices = [];

const editorNoticeKey = ( postID ) => {
	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
};

const loadIdeaHubNotices = async ( _global = global ) => {
	const { wp } = _global;

	const hasNotice = ( postID ) => {
		if ( wp.data.select( 'core/notices' ).getNotices() === undefined ) {
			return undefined;
		}

		return wp.data.select( 'core/notices' ).getNotices().some( ( notice ) => {
			return notice.id === editorNoticeKey( postID );
		} );
	};

	const listener = async () => {
		// eslint-disable-next-line sitekit/acronym-case
		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();

		if ( ! postID ) {
			return;
		}

		const { cacheHit } = await getItem( editorNoticeKey( postID ) );

		if ( hasNotice( postID ) === false ) {
			// We've already shown this notice, so when it's hidden, mark it as shown
			// so it doesn't appear again.
			if ( shownNotices.includes( editorNoticeKey( postID ) ) ) {
				setItem(
					editorNoticeKey( postID ),
					postMeta.googlesitekitpersistent_idea_text,
					{ ttl: 108000 } // Don't show this notice for another 90 days.
				);
				unsubscribeFromListener();
				return;
			}

			// We've already shown this notice on a previous visit to this page in
			// the editor, so don't show it again.
			if ( cacheHit ) {
				unsubscribeFromListener();
				return;
			}

			// We haven't shown any notice for this post before, so let's check for
			// Idea Hub postmeta.
			const postMeta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );

			// eslint-disable-next-line camelcase
			if ( postMeta?.googlesitekitpersistent_idea_text ) {
				wp.data.dispatch( 'core/notices' ).createInfoNotice(
					/* translators: %s: Idea post name */
					sprintf( __( 'This post was created from an idea you picked in Site Kitâ€™s Idea Hub: %s', 'google-site-kit' ), postMeta.googlesitekitpersistent_idea_text ),
					{ id: editorNoticeKey( postID ) }
				);

				// Mark the notice as shown locally, so if it is no longer marked as
				// visible, we known it was dismissed (rather than having yet to be
				// displayed).
				shownNotices.push( editorNoticeKey( postID ) );
			}
		}
	};

	const unsubscribeFromListener = wp.data.subscribe( listener );
};

loadIdeaHubNotices();
