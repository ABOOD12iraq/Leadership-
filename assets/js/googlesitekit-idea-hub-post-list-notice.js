/**
 * Idea Hub Post List notice.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
// import './modules';

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
// import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
const { dispatch } = Data;
import { CORE_USER } from './googlesitekit/datastore/user/constants';

const WEEK_IN_SECONDS = 3600 * 24 * 7;

const dismissNotice = ( slug, expirationTime ) => dispatch( CORE_USER ).dismissItem( slug, expirationTime );

domReady( () => {
	const newNotice = document.getElementById( 'googlesitekit-notice-new' );
	const savedNotice = document.getElementById( 'googlesitekit-notice-saved' );
	let type, notice;
	if ( newNotice ) {
		type = 'new';
		notice = newNotice;
	} else if ( savedNotice ) {
		type = 'saved';
		notice = savedNotice;
	} else {
		return;
	}

	// Button pops up only after the timeout passes
	setTimeout( () => {
		const button = notice.getElementsByClassName( 'notice-dismiss' )[ 0 ];
		const slug = type === 'new' ? 'new-ideas' : 'saved-ideas';
		const expirationType = type === 'new' ? 0 : WEEK_IN_SECONDS;
		button.addEventListener( 'click', () => dismissNotice( slug, expirationType ) );
	}, 1 );
} );
