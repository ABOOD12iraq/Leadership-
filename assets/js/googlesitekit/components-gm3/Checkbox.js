/**
 * Checkbox component.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@material/web/checkbox/checkbox';
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { useEffect, useRef } from '@wordpress/element';
import Spinner from '../../components/Spinner';

export default function Checkbox( {
	onChange,
	id,
	name,
	value,
	checked,
	disabled,
	children,
	tabIndex,
	onKeyDown,
	loading,
} ) {
	const ref = useRef( null );

	const controlledState = useRef( {} );
	controlledState.current.checked = checked;

	useEffect( () => {
		const { current } = ref;

		if ( ! current ) {
			return;
		}

		const change = ( event ) => {
			onChange?.( event );

			current.checked = controlledState.current.checked;

			// The checked property doesn't get reflected to the inner input element checked state,
			// so we need to set it manually.
			current.shadowRoot.querySelector( 'input' ).checked =
				controlledState.current.checked;
		};

		current.addEventListener( 'change', change );

		if ( onKeyDown ) {
			current.addEventListener( 'keydown', onKeyDown );
		}

		return () => {
			current.removeEventListener( 'change', change );

			if ( onKeyDown ) {
				current.removeEventListener( 'keydown', onKeyDown );
			}
		};
	}, [ checked, disabled, onChange, onKeyDown ] );

	useEffect( () => {
		const { current } = ref;

		if ( ! current ) {
			return;
		}

		current.checked = checked;

		// The checked property doesn't get reflected to the inner input element checked state,
		// so we need to set it manually.
		const innerInput = current.shadowRoot.querySelector( 'input' );
		if ( innerInput ) {
			innerInput.checked = checked;
		}
	}, [ checked ] );

	return (
		<div className="googlesitekit-component-gm3__checkbox">
			{ loading && (
				<div className="googlesitekit-component-gm3__checkbox--loading">
					<Spinner isSaving />
				</div>
			) }
			{ ! loading && (
				<md-checkbox
					id={ id }
					ref={ ref }
					role="checkbox"
					// Lit boolean attributes treat anything non-null|undefined as true. Coerce to null if false.
					// See https://lit.dev/docs/v1/components/properties/#attributes
					checked={ checked || null }
					disabled={ disabled || null }
					name={ name }
					value={ value }
					tabIndex={ tabIndex }
				></md-checkbox>
			) }
			<label htmlFor={ id }>{ children }</label>
		</div>
	);
}

Checkbox.propTypes = {
	onChange: PropTypes.func.isRequired,
	onKeyDown: PropTypes.func,
	id: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	checked: PropTypes.bool,
	disabled: PropTypes.bool,
	children: PropTypes.node.isRequired,
	tabIndex: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ),
	loading: PropTypes.bool,
};

Checkbox.defaultProps = {
	checked: false,
	disabled: false,
	tabIndex: undefined,
	onKeyDown: null,
	loading: false,
};
