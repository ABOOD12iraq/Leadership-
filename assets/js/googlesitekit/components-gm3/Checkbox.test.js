/**
 * Checkbox tests.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { useState } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { fireEvent, render } from '../../../../tests/js/test-utils';
import Checkbox from './Checkbox';

// import '@material/web/checkbox/checkbox';
// function Checkbox() {
// 	return <md-checkbox></md-checkbox>;
// }

// TODO: Worth a look - https://www.npmjs.com/package/shadow-dom-testing-library

describe( 'Checkbox', () => {
	it( 'should render the checkbox', () => {
		const { container } = render( <Checkbox /> );
		expect( container ).toMatchSnapshot();
	} );

	describe( 'controlled input behaviour', () => {
		it( 'should correctly invoke onChange and retain its unchecked state when clicked', () => {
			const onChange = jest.fn();
			const { container, getByRole } = render(
				<Checkbox onChange={ onChange } />
			);

			expect( container ).toMatchSnapshot();

			// Confirm the checkbox is not checked.
			expect( getByRole( 'checkbox' ) ).not.toHaveAttribute( 'checked' );

			fireEvent.click( getByRole( 'checkbox' ) );

			expect( container ).toMatchSnapshot();

			// Confirm the click resulted in a change event with target.checked: true.
			expect( onChange ).toHaveBeenCalledTimes( 1 );
			const event = onChange.mock.calls[ 0 ][ 0 ];
			expect( event.target.checked ).toBe( true );

			// Confirm the checkbox remains unchecked as its state is controlled.
			expect( getByRole( 'checkbox' ) ).not.toHaveAttribute( 'checked' );
		} );

		it( 'should allow updating of the checked state', () => {
			function CheckableCheckbox( { onChange } ) {
				const [ checked, setChecked ] = useState( false );

				return (
					<Checkbox
						checked={ checked }
						onChange={ ( event ) => {
							onChange( event );
							setChecked( event.target.checked );
						} }
					/>
				);
			}

			const onChange = jest.fn();
			const { container, getByRole } = render(
				<CheckableCheckbox onChange={ onChange } />
			);

			expect( container ).toMatchSnapshot();

			// Confirm the checkbox is not checked.
			expect( getByRole( 'checkbox' ) ).not.toHaveAttribute( 'checked' );

			fireEvent.click( getByRole( 'checkbox' ) );

			expect( container ).toMatchSnapshot();

			// Confirm the click resulted in a change event with target.checked: true.
			expect( onChange ).toHaveBeenCalledTimes( 1 );
			let event = onChange.mock.calls[ 0 ][ 0 ];
			expect( event.target.checked ).toBe( true );

			// Confirm the checkbox is now checked
			expect( getByRole( 'checkbox' ) ).toHaveAttribute( 'checked' );

			// Click again to uncheck
			onChange.mockClear();
			fireEvent.click( getByRole( 'checkbox' ) );

			expect( container ).toMatchSnapshot();

			// Confirm the click resulted in a change event with target.checked: undefined.
			expect( onChange ).toHaveBeenCalledTimes( 1 );
			event = onChange.mock.calls[ 0 ][ 0 ];
			expect( event.target ).toHaveProperty( 'checked', undefined );

			// Confirm the checkbox is now unchecked
			expect( getByRole( 'checkbox' ) ).not.toHaveAttribute( 'checked' );
		} );
	} );
} );
