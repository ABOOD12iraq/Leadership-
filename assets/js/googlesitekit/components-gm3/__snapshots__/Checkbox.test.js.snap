// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox controlled input behaviour should allow updating of the checked state 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should allow updating of the checked state 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      checked=""
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should allow updating of the checked state 3`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should correctly invoke onChange and retain its checked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      checked=""
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should correctly invoke onChange and retain its checked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      checked=""
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should correctly invoke onChange and retain its unchecked state when clicked 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox controlled input behaviour should correctly invoke onChange and retain its unchecked state when clicked 2`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;

exports[`Checkbox should render the checkbox 1`] = `
<div>
  <div
    class="googlesitekit-component-gm3__checkbox"
  >
    <md-checkbox
      id="checkbox-id"
      name="checkbox-name"
      role="checkbox"
      value="checkbox-value"
    />
    <label
      for="checkbox-id"
    >
      Checkbox Label
    </label>
  </div>
</div>
`;
